src/Core/Math/Ellipsoid.js

```diff
@@ -1,182 +1,171 @@
-/**
- * Generated On: 2015-10-5
- * Class: Ellipsoid
- * Description: Classe mathématique de  l'ellispoide
- */
-
-
 import * as THREE from 'three';
+import proj4 from 'proj4';
 import Coordinates from 'Core/Geographic/Coordinates';
 
-function Ellipsoid(size) {
-    // Constructor
-
+export const ellipsoidSizes = {
+    x: proj4.WGS84.a,
+    y: proj4.WGS84.a,
+    z: proj4.WGS84.b,
+};
 
-    this.rayon_1 = size.x;
-    this.rayon_2 = size.y;
-    this.rayon_3 = size.z;
+const normal = new THREE.Vector3();
 
-    this.size = new THREE.Vector3(size.x, size.y, size.z);
+class Ellipsoid {
+    constructor(size = ellipsoidSizes) {
+        this.rayon_1 = size.x;
+        this.rayon_2 = size.y;
+        this.rayon_3 = size.z;
 
-    this._radiiSquared = new THREE.Vector3(size.x * size.x, size.y * size.y, size.z * size.z);
+        this.size = new THREE.Vector3(size.x, size.y, size.z);
 
-    this._oneOverRadiiSquared = new THREE.Vector3(size.x === 0.0 ? 0.0 : 1.0 / (size.x * size.x),
-        size.y === 0.0 ? 0.0 : 1.0 / (size.y * size.y),
-        size.z === 0.0 ? 0.0 : 1.0 / (size.z * size.z));
-}
+        this._radiiSquared = new THREE.Vector3(size.x * size.x, size.y * size.y, size.z * size.z);
 
-Ellipsoid.prototype.geodeticSurfaceNormal = function geodeticSurfaceNormal(cartesian, target = new THREE.Vector3()) {
-    target.set(
-        cartesian.x() * this._oneOverRadiiSquared.x,
-        cartesian.y() * this._oneOverRadiiSquared.y,
-        cartesian.z() * this._oneOverRadiiSquared.z);
-    return target.normalize();
-};
+        this._oneOverRadiiSquared = new THREE.Vector3(size.x === 0.0 ? 0.0 : 1.0 / (size.x * size.x),
```

Comment:
`this._oneOverRadiiSquared = new THREE.Vector3(1,1,1).divide(this._radiiSquared);`

Reply:
I couldn't really change that: see my updated proposal.